src/thriveopedia/fossilisation/FossilisationDialog.cs

```diff
@@ -0,0 +1,232 @@
+    {
+        if (valid)
+        {
+            GUICommon.MarkInputAsValid(speciesNameEdit);
+            fossiliseButton.Disabled = false;
+
+            SelectedSpecies.UpdateNameIfValid(speciesNameEdit.Text);
+        }
+        else
+        {
+            GUICommon.MarkInputAsInvalid(speciesNameEdit);
+            fossiliseButton.Disabled = true;
+        }
+    }
+
+    public void OnClickedOffName()
+    {
+        var focused = GetFocusOwner();
+
+        // Ignore if the species name line edit wasn't focused or if one of our controls is hovered
+        if (focused != speciesNameEdit || controlsHoveredOver)
+            return;
+
+        PerformValidation(speciesNameEdit.Text);
+    }
+
+    private void OnNameTextChanged(string newText)
+    {
+        ReportValidityOfName(Regex.IsMatch(newText, Constants.SPECIES_NAME_REGEX));
+    }
+
+    private void OnNameTextEntered(string newText)
+    {
+        PerformValidation(newText);
+    }
+
+    private void PerformValidation(string text)
+    {
+        // Only defocus if the name is valid to indicate invalid namings to the player
+        if (Regex.IsMatch(text, Constants.SPECIES_NAME_REGEX))
+        {
+            speciesNameEdit.ReleaseFocus();
+        }
+        else
+        {
+            // Prevents user from doing other actions with an invalid name
+            GetTree().SetInputAsHandled();
+
+            // TODO: Make the popup appear at the top of the line edit instead of at the last mouse position
+            ToolTipManager.Instance.ShowPopup(TranslationServer.Translate("INVALID_SPECIES_NAME_POPUP"), 2.5f);
+
+            speciesNameEdit.GetNode<AnimationPlayer>("AnimationPlayer").Play("invalidSpeciesNameFlash");
+        }
+    }
+
+    private void OnRandomizeNamePressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        var nameGenerator = SimulationParameters.Instance.NameGenerator;
+        var randomizedName = nameGenerator.GenerateNameSection() + " " +
+            nameGenerator.GenerateNameSection(null, true);
+
+        speciesNameEdit.Text = randomizedName;
+        OnNameTextChanged(randomizedName);
+    }
+
+    private void UpdateSpeciesPreview()
+    {
+        speciesPreview.PreviewSpecies = SelectedSpecies;
+
+        switch (SelectedSpecies)
+        {
+            case MicrobeSpecies microbeSpecies:
+            {
+                hexesPreview.PreviewSpecies = microbeSpecies;
+                break;
+            }
+        }
+    }
+
+    private void UpdateSpeciesDetails()
+    {
+        speciesDetailsLabel.ExtendedBbcode = TranslationServer.Translate("SPECIES_DETAIL_TEXT").FormatSafe(
+            SelectedSpecies.FormattedNameBbCode,
+            SelectedSpecies.ID,
+            SelectedSpecies.Generation,
+            SelectedSpecies.Population,
+            SelectedSpecies.Colour.ToHtml(),
+            string.Join("\n  ", SelectedSpecies.Behaviour.Select(b => b.Key + ": " + b.Value)));
```

Comment:
Wasn't there behaviour value translations in the auto-evo exploring tool?

Reply:
Or is this code exactly the same as in there? If so this is pretty complex so I'd say this should be made into a helper method somewhere (probably the Species class makes most sense).